tests/test_task.py

```diff
@@ -0,0 +1,132 @@
+"""
+Task Test Suite
+"""
+
+import unittest
+
+from pydantic import Field, ValidationInfo, field_validator
+
+import aind_behavior_curriculum as abc
+
+
+class ExampleTask(abc.Task):
+    """
+    Example Task
+    """
+
+    # Required: Define type annotations for strict type checks.
+    # Optional: Make fields immutable with the 'frozen=True'.
+    field_1: int = Field(0)
+    field_2: int = Field(0)
+    field_3: float = Field(0.5)
+    field_4: float = Field(0.5)
+    field_5: str = Field("Immutable Field", frozen=True)
+
+    # Optional: Add additional validation to fields.
+    @field_validator("field_1", "field_2")
+    @classmethod
+    def check_nonnegative(cls, v: int, info: ValidationInfo):
```

Comment:
Again, nothing wrong with this, and i think it is an amazing example. But it is one of those cases where the validation could be simply coded in the `Field` method using the `ge` keyword. This validation would be provided for free by the json-schema. I do agree that there will be validation that wont be possible to be made at the json-schema level, but when it is, I would favor it since pydantic will "inherit" it anyway.

Reply:
Updated